### ISSUE TITLE
ERR_CONNECTION_REFUSED starting Electron example

### ISSUE THREAD

**Issue by MarkZ3** [theia-ide/theia#308]
npm 4.2.0
node 7.10.0
macOS 10.12.5 and similar on Ubuntu 16.04
Latest master (I did a git clean -fdx)

I did npm run start in examples/electron. When it starts, I get a lot of errors in the DevTools log and the Files navigator doesn't work.

(Both macOS and Ubuntu)
```
index.js:117 WebSocket connection to 'ws://127.0.0.1:3000/services/filesystem' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
connect @ index.js:117
ReconnectingWebsocket @ index.js:144
WebSocketConnectionProvider.createWebSocket @ connection.ts:87
WebSocketConnectionProvider.listen @ connection.ts:48
WebSocketConnectionProvider.createProxy @ connection.ts:36
WebSocketConnectionProvider.createProxy @ connection.ts:24
(anonymous) @ filesystem-frontend-module.ts:30
_resolveRequest @ resolver.js:40
(anonymous) @ instantiation.js:42
resolveInstance @ instantiation.js:41
_resolveRequest @ resolver.js:49
(anonymous) @ resolver.js:17
_resolveRequest @ resolver.js:16
resolve @ resolver.js:66
(anonymous) @ container.js:230
Container._get @ container.js:221
Container.getAll @ container.js:189
ContainerBasedContributionProvider.getContributions @ contribution-provider.ts:29
CommandRegistry.onStart @ command.ts:49
FrontendApplication.startContributions @ frontend-application.ts:86
FrontendApplication.start @ frontend-application.ts:58

```

In the Terminal (macOS), I can also see this:

```
Error: The module '/Users/mark/git/theia/node_modules/dtrace-provider/build/Release/DTraceProviderBindings.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 51. This version of Node.js requires
NODE_MODULE_VERSION 53. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or`npm install`).
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:173:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:173:20)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/mark/git/theia/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Object.<anonymous> (/Users/mark/git/theia/node_modules/dtrace-provider/dtrace-provider.js:42:3)
{ Error: Cannot find module './build/default/DTraceProviderBindings'
    at Module._resolveFilename (module.js:470:15)
    at Function.Module._resolveFilename (/Users/mark/git/theia/node_modules/electron/dist/Electron.app/Contents/Resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/mark/git/theia/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Object.<anonymous> (/Users/mark/git/theia/node_modules/dtrace-provider/dtrace-provider.js:42:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32) code: 'MODULE_NOT_FOUND' }
{ Error: Cannot find module './build/Debug/DTraceProviderBindings'
    at Module._resolveFilename (module.js:470:15)
    at Function.Module._resolveFilename (/Users/mark/git/theia/node_modules/electron/dist/Electron.app/Contents/Resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/mark/git/theia/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Object.<anonymous> (/Users/mark/git/theia/node_modules/dtrace-provider/dtrace-provider.js:42:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32) code: 'MODULE_NOT_FOUND' }
(node:35746) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: The module '/Users/mark/git/theia/node_modules/node-pty/build/Release/pty.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 51. This version of Node.js requires
NODE_MODULE_VERSION 53. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or`npm install`).
(node:35746) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
2017-07-18 12:07:52.325 Electron Helper[35749:1067115] Couldn't set selectedTextBackgroundColor from default ()
```

On Ubuntu:

```
(node:6859) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: The module '/home/emalape/git/theia/node_modules/node-pty/build/Release/pty.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 51. This version of Node.js requires
NODE_MODULE_VERSION 53. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or`npm install`).
(node:6859) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by akosyakov** [comment]
@MarkZ3 Could you try with #310?

**Comment by MarkZ3** [comment]
It seems to work much better now, no ERR_CONNECTION_REFUSED and the Navigator works. I have this in the DevTools console thought, not sure how bad it is. This is on macOS

```
bundle.js:20495 Uncaught TypeError: Cannot assign to read only property 'exports' of object '#<Object>'
    at Object.<anonymous> (bundle.js:20495)
    at Object.defineProperty.value (bundle.js:20515)
    at __webpack_require__ (bundle.js:653)
    at Object.defineProperty.value (bundle.js:742)
    at bundle.js:745
    at webpackUniversalModuleDefinition (bundle.js:7)
    at bundle.js:10
```

Oddly enough, on this branch on Ubuntu, now when I do "npm run start" in examples/electron, I get:
```
> @theia/example-electron@0.0.0 build:backend /home/mark/git/theia/examples/electron
> cp src-gen/frontend/index.html lib

/home/mark/git/theia/node_modules/electron/index.js:9
  throw new Error('Electron failed to install correctly, please delete node_modules/electron and try installing again')
  ^

Error: Electron failed to install correctly, please delete node_modules/electron and try installing again
    at Object.<anonymous> (/home/mark/git/theia/node_modules/electron/index.js:9:9)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/mark/git/theia/node_modules/electron/cli.js:3:16)
    at Module._compile (module.js:571:32)

```

**Comment by akosyakov** [comment]
Could you try to add `'--no-insight'` as args in scripts/generate.js and run again:
`const args = ['theia:' + target, '--force', '--no-insight']`

**Comment by akosyakov** [comment]
I will try to reproduce and fix this issue: Uncaught TypeError: Cannot assign to read only property 'exports' of object '#<Object>'

Ubuntu: never seen such error.

**Comment by hexa00** [comment]
I can reproduce that too, maybe be linked with this : https://github.com/webpack/webpack/issues/4039

**Comment by akosyakov** [comment]
@MarkZ3 @hexa00 Could you check with https://github.com/theia-ide/theia/pull/319. There an error is gone. The thing is that if one uses es6 imports then he should use exports, `module.exports =` works only with es5.

**Comment by MarkZ3** [comment]
@akosyakov with #319 I don't see the error anymore. I think this one can be closed.

**Comment by akosyakov** [comment]
The last error is fixed with #319
